yrs/src/block.rs

```diff
@@ -1137,12 +1138,12 @@ impl SplittableString {
     fn map_utf16_offset(&self, offset: u32) -> u32 {
         let mut off = 0;
         let mut i = 0;
-        for c in self.content.encode_utf16() {
+        for c in self.content.chars() {
             if i >= offset {
                 break;
             }
-            off += if c < 0x80 { 1 } else { 2 };
-            i += 1;
+            off += c.len_utf8() as u32;
+            i += c.len_utf16() as u32;
```

Comment:
Offset provided in this method is UTF-16 aligned (since its unit of measure for Yjs `item.id.clock`), while we need to convert it to UTF-8 aligned (since it's the index for Rust's string splitting).